test/puppeteer/demo.test.ts

```diff
@@ -72,6 +72,23 @@ it('closes the menu if you click outside of it', async () => {
 	expect(true).toBe(true);
 });
 
+it('closes the menu if you click a menu item with a click handler', async () => {
```

Comment:
This test never actually clicks a menu item; it simply uses the `Enter` key to activate an item with the keyboard. Also, is it necessary to click the third item, where we have to add logic to dismiss the dialog?

Reply:
Good catch,  I will change it to a click. We do  need to click a menu item that has a callback, as that is the scenario we need to manage the menu automatically closing in. If the user clicks a link the menu will automatically close since the whole component will dismount, and if they command + click it, we do not want the menu to close. I could add another menu item that does not bring up a dialog, but still calls a function, if that's preferable.

Reply:
> since the whole component will dismount

This isn't always the case with every link. Some links, like ones from a SPA routing framework like React Router, won't actually navigate to a new page.

Reply:
The mention of command clicking links is interesting. It sheds light on the fact that there are probably many situations where we don't want the menu to automatically close on a click.

I'm wondering if, instead of automatically closing it, we provide developers with a way to close the menu programmatically. Then it'd be up to the developer to close it when they think it's best.

Reply:
We could return the state setting function for isOpen. That would give the developer the ability to close or open the menu in any cases where they needed to for their implementation. If we are going to add that to the list of returned values, perhaps we should be returning an object instead of an array.

Reply:
I think that's best; that would give the dev full control over when it should close. Although, it should still close automatically upon activating an item with the keyboard, since that's in keeping with the WAI-ARIA guidelines.

Reply:
Do you want to update this PR to do this?

Reply:
Yes I will handle that in this PR.